pkg/provision/storage/perWorkspaceStorage_test.go

```diff
@@ -0,0 +1,105 @@
+//
+// Copyright (c) 2019-2022 Red Hat, Inc.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package storage
+
+import (
+	"testing"
+
+	dw "github.com/devfile/api/v2/pkg/apis/workspaces/v1alpha2"
+	"github.com/devfile/devworkspace-operator/pkg/common"
+	"github.com/devfile/devworkspace-operator/pkg/provision/sync"
+	"github.com/google/go-cmp/cmp"
+
+	"github.com/stretchr/testify/assert"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	clientgoscheme "k8s.io/client-go/kubernetes/scheme"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"
+
+	"github.com/devfile/devworkspace-operator/apis/controller/v1alpha1"
+	"github.com/devfile/devworkspace-operator/pkg/config"
+)
+
+func init() {
+	utilruntime.Must(clientgoscheme.AddToScheme(scheme))
+	utilruntime.Must(v1alpha1.AddToScheme(scheme))
+	utilruntime.Must(dw.AddToScheme(scheme))
+	config.SetConfigForTesting(nil)
+}
+
+func TestRewriteContainerVolumeMountsForPerWorkspaceStorageClass(t *testing.T) {
+	tests := loadAllTestCasesOrPanic(t, "testdata/perWorkspace-storage")
+	setupControllerCfg()
+	perWorkspaceStorage := perWorkspaceStorageProvisioner{}
+
+	for _, tt := range tests {
+		t.Run(tt.Name, func(t *testing.T) {
+			// sanity check that file is read correctly.
+			assert.NotNil(t, tt.Input.Workspace, "Input does not define workspace")
+			workspace := &dw.DevWorkspace{}
+			workspace.Spec.Template = *tt.Input.Workspace
+			workspace.Status.DevWorkspaceId = tt.Input.DevWorkspaceID
+			workspace.Namespace = "test-namespace"
+
+			clusterAPI := sync.ClusterAPI{
+				Scheme: scheme,
+				Client: fake.NewClientBuilder().WithScheme(scheme).Build(),
+				Logger: zap.New(),
+			}
+
+			if needsStorage(&workspace.Spec.Template) {
+				err := perWorkspaceStorage.ProvisionStorage(&tt.Input.PodAdditions, workspace, clusterAPI)
+				if !assert.Error(t, err, "Should get an error when creating PVC") {
```

Comment:
Minor edit for clarity:
```suggestion
				if !assert.Error(t, err, "Should get a NotReady error when creating PVC") {
```